packages/rum-core/src/boot/preStartRum.ts

```diff
@@ -0,0 +1,169 @@
+      initialViewOptions = firstStartView!.options
+    }
+
+    const startRumResult = doStartRum(cachedInitConfiguration, cachedConfiguration, deflateWorker, initialViewOptions)
+
+    bufferApiCalls.drain(startRumResult)
+  }
+
+  return {
+    init(initConfiguration) {
+      if (!initConfiguration) {
+        display.error('Missing configuration')
+        return
+      }
+
+      const eventBridgeAvailable = canUseEventBridge()
+      if (eventBridgeAvailable) {
+        initConfiguration = overrideInitConfigurationForBridge(initConfiguration)
+      }
+
+      // Expose the initial configuration regardless of initialization success.
+      cachedInitConfiguration = initConfiguration
+
+      if (cachedConfiguration) {
+        displayAlreadyInitializedError('DD_RUM', initConfiguration)
+        return
+      }
+
+      // If we are in a Synthetics test configured to automatically inject a RUM instance, we want to
+      // completely discard the customer application RUM instance by ignoring their init() call.  But,
+      // we should not ignore the init() call from the Synthetics-injected RUM instance, so the
+      // internal `ignoreInitIfSyntheticsWillInjectRum` option is here to bypass this condition.
+      if (ignoreInitIfSyntheticsWillInjectRum && willSyntheticsInjectRum()) {
+        return
+      }
+
+      const configuration = validateAndBuildRumConfiguration(initConfiguration)
+      if (!configuration) {
+        return
+      }
+
+      if (!eventBridgeAvailable && !configuration.sessionStoreStrategyType) {
+        display.warn('No storage available for session. We will not send any data.')
+        return
+      }
+
+      if (configuration.compressIntakeRequests && !eventBridgeAvailable && startDeflateWorker) {
+        deflateWorker = startDeflateWorker(
+          configuration,
+          'Datadog RUM',
+          // Worker initialization can fail asynchronously, especially in Firefox where even CSP
+          // issues are reported asynchronously. For now, the SDK will continue its execution even if
+          // data won't be sent to Datadog. We could improve this behavior in the future.
+          noop
+        )
+        if (!deflateWorker) {
+          // `startDeflateWorker` should have logged an error message explaining the issue
+          return
+        }
+      }
+
+      cachedConfiguration = configuration
+      tryStartRum()
+    },
+
+    get initConfiguration() {
+      return cachedInitConfiguration
+    },
+
+    getInternalContext: noop as () => undefined,
+
+    stopSession: noop,
+
+    addTiming(name, time = timeStampNow()) {
+      bufferApiCalls.add((startRumResult) => startRumResult.addTiming(name, time))
+    },
+
+    startView(options, startClocks = clocksNow()) {
+      let ignore = false
+      if (!firstStartView) {
+        firstStartView = {
+          options,
+          ignore() {
+            ignore = true
+          },
+        }
+        tryStartRum()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what is happening around this first start view ignore?

Reply:
With `trackViewManually: true`, we want to use the first `startView()` call options, and don't actually call `startView`. This is the same behavior as before, but a bit more explicit.

Reply:
ok, I get it now.
What about:
- add an extra comment to also explain it here
- rename `firstStartView` to something like `firstStartViewCall`, to make a bit more explicit that we talk about the API call

Reply:
Since Aymeric also had some suggestions on that part, I changed the logic a little bit. Let me know WDYT :)